kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -105,8 +107,11 @@ public Transaction build(Engine engine) {
       // Table doesn't exist yet. Create an initial snapshot with the new schema.
       Metadata metadata = getInitialMetadata();
       Protocol protocol = getInitialProtocol();
-      LogReplay logReplay = getEmptyLogReplay(engine, metadata, protocol);
-      snapshot = new InitialSnapshot(table.getDataPath(), logReplay, metadata, protocol);
+      SnapshotContext snapshotContext = SnapshotContext.forVersionSnapshot(tablePath, 0);
```

Comment:
I updated this, my mistake this should've been -1 (which matches the version returned by the log segment & log replay).

In general it is the intended read version. For this specific case, we treat a non-existent table as read version -1. Note, we don't generate a snapshot report for this case since there is no real snapshot to load!